build.gradle

```diff
@@ -1,114 +1,147 @@
 def railsVendor = 'Rails project'
-def railsHomepage = 'http://rails.sf.net'
+//def railsHomepage = 'http://rails.sf.net'
 def railsDescription = 'Rails is a moderator program for 18xx games'
 def railsIcon = 'rails.ico'
 def railsCopyright = 'GPL'
 
 // Key variables
-def keyAlias = 'rails18xx'
+//def keyAlias = 'rails18xx'
 // def keyAlias = 'stefanfrey'
 
 // Build directories
-def webstartBuildDir = "${buildDir}/webstart/build"
-def webstartDestDir = "${buildDir}/webstart/signed"
-def keystoreFile = '../../keystore/rails.keys'
+//def webstartBuildDir = "${buildDir}/webstart/build"
+//def webstartDestDir = "${buildDir}/webstart/signed"
+//def keystoreFile = '../../keystore/rails.keys'
 def buildFilesDir = 'src/main/resources/build'
-def resourceFilesDir = 'src/main/resources'
-def versionFile = "version.number"
+//def resourceFilesDir = 'src/main/resources'
+//def versionFile = "version.number"
 
 // Sourceforge variables
-def sourceforgeJnlpPath = "http://${railsCodebase}/webstart"
-def sourceforgeUploadPath = 'web.sourceforge.net/home/project-web/rails/htdocs/webstart'
-def sourceforgeJnlpFileName = 'rails.jnlp'
-def sourceforgeUserName = 'stefanfrey'
+//def sourceforgeJnlpPath = "http://${railsCodebase}/webstart"
+//def sourceforgeUploadPath = 'web.sourceforge.net/home/project-web/rails/htdocs/webstart'
+//def sourceforgeJnlpFileName = 'rails.jnlp'
+//def sourceforgeUserName = 'stefanfrey'
 
-// use version number from property file
-propFileLoader.load(file("${buildFilesDir}/${versionFile}"))
+dependencies {
+    implementation 'commons-io:commons-io:2.6'
+    implementation 'com.github.vlsi.mxgraph:jgraphx:4.1.0'
+    implementation 'org.jgrapht:jgrapht-core:1.4.0'
+    implementation 'org.jgrapht:jgrapht-ext:1.4.0'
+    implementation 'org.slf4j:slf4j-api:1.7.30'
+    implementation 'org.apache.xmlgraphics:batik-bridge:1.12'
+    implementation 'org.apache.xmlgraphics:batik-transcoder:1.12'
+    implementation 'org.apache.xmlgraphics:batik-svggen:1.12'
+    implementation 'org.apache.xmlgraphics:batik-ext:1.12'
+    implementation 'org.apache.xmlgraphics:batik-swing:1.12'
+    implementation 'javazoom:jlayer:1.0.1'
+    implementation 'ch.qos.logback:logback-classic:1.2.3'
+    implementation 'com.google.guava:guava:23.0'
+    implementation 'net.java.dev.designgridlayout:designgridlayout:1.11'
+    implementation 'org.dockingframes:docking-frames-common:1.1.1'
+    implementation 'com.googlecode.combinatoricslib:combinatoricslib:2.2'
+    //implementation 'org.openjfx:javafx-controls:14'
+    implementation 'org.apache.httpcomponents:httpclient:4.5.12'
+    implementation 'org.apache.commons:commons-lang3:3.9'
+
+    testImplementation 'junit:junit:4.13'
+    testImplementation 'org.easytesting:fest-assert-core:2.0M10'
+    testImplementation 'org.easytesting:fest-util:1.2.5'
+    testImplementation 'org.mockito:mockito-core:3.3.3'
+}
 
+javafx {
+    version = "14"
+    modules = [ 'javafx.controls' ]
+}
```

Comment:
We should also check how to include all native jars (i.e. for linux, mac and windows) inside our `jar` bundles. Instead of limiting it to the one for the current os. I'm not sure if the `org.openjfx.javafxplugin` plugin does this automatically.

Reply:
It would need to be a part of the bundling process. See section 4 here:
* https://github.com/openjfx/javafx-gradle-plugin

Reply:
Was thinking about this more-- given we should really be building artifacts from CI, maybe we don't really need to worry about this? Maybe set up CI to build the artifact for a respective platform FROM that respective platform? What platform options can we build on? Guessing Mac is tough but what about the others?

Reply:
> Maybe set up CI to build the artifact for a respective platform FROM that respective platform?

That is exactly the idea I was going for with #114 ;)

> What platform options can we build on? Guessing Mac is tough but what about the others?

We can build on Windows, Mac and Ubuntu. `jpackage` supports the formats: 

```
    Linux: deb and rpm
    macOS: pkg and dmg
    Windows: msi and exe
```

(as taken from https://openjdk.java.net/jeps/343)